eureka-client/src/main/java/com/netflix/appinfo/ApplicationInfoManager.java

```diff
@@ -219,6 +219,21 @@ public void refreshDataCenterInfoIfRequired() {
             builder.setHostName(newAddress).setIPAddr(newIp).setDataCenterInfo(config.getDataCenterInfo());
             instanceInfo.setIsDirty();
         }
+
+        String existingSpotInstanceAction  = instanceInfo.getMetadata().get(AmazonInfo.MetaDataKey.spotInstanceAction.name);
+        DataCenterInfo dataCenterInfo = config.getDataCenterInfo();
+        if (dataCenterInfo instanceof AmazonInfo) {
+            AmazonInfo amazonInfo = (AmazonInfo) dataCenterInfo;
+            String newSpotInstanceAction = amazonInfo.getMetadata().get(AmazonInfo.MetaDataKey.spotInstanceAction.name);
```

Comment:
It is unfortunate that the API for `AmazonInfo` is a bit leaky, but the preferred API to extract data is
https://github.com/Netflix/eureka/blob/f8439886921e3655a301adb149a13eaa34268074/eureka-client/src/main/java/com/netflix/appinfo/AmazonInfo.java#L299, where the enum can be used directly.

Reply:
Got it. Fixed.